### PR TITLE
Remove AppAuth dependency

### PR DESCRIPTION
**Pull request by ps2**
Same functionality.  A lot less code.

Also includes some fixes to match Tidepool API attribute names: `types` -> `type`, and `subTypes` -> `subType`

### CODE REVIEW

File: Sources/TidepoolKit/Auth/OAuth2Authenticator.swift [hunk 1]
```diff
@@ -9,19 +9,53 @@
...
+                    continuation.resume(throwing: TError.missingAuthenticationState)
+                    return
+                }
+
+                continuation.resume(returning: callbackURL)
+            }
+            authenticationSession?.presentationContextProvider = contextProviding
+            authenticationSession?.start()
+        }
+        return callbackURL
```
Comment (Camji55) [inline Sources/TidepoolKit/Auth/OAuth2Authenticator.swift]: Did you need to declare the variable and type here to satisfy the compiler? I know sometimes it's finicky with the `continuations`. I think you might be able to get around that with this:
```swift
public func startSession(authURL: URL, callbackScheme: String?) async throws -> URL {
    try await withCheckedThrowingContinuation<URL> { continuation in
        ...
    }
}
```

File: Sources/TidepoolKit/TAPI.swift [hunk 2]
```diff
@@ -296,35 +232,47 @@ public actor TAPI {
 
         let config = try await lookupOIDConfiguration(environment: session.environment)
 
-        guard let oidConfig = OIDServiceConfiguration(from: config) else {
+        var components = URLComponents()
+        components.queryItems = [
+            URLQueryItem(name:"refresh_token", value: refreshToken),
+            URLQueryItem(name:"grant_type", value: "refresh_token"),
```
Comment (Camji55) [inline Sources/TidepoolKit/TAPI.swift:238]: Just checking this is correct?

**Review by Camji55 — APPROVED** [review]

File: Sources/TidepoolKit/TAPI.swift (hunk 2 above)
Comment (ps2) [reply Sources/TidepoolKit/TAPI.swift:238]: Are you asking about the `grant_type`?  If so, yes, "refresh_token" is the correct grant type when asking to refresh the access token.

File: Sources/TidepoolKit/Auth/OAuth2Authenticator.swift (hunk 1 above)
Comment (ps2) [reply Sources/TidepoolKit/Auth/OAuth2Authenticator.swift]: Ah, yeah, thanks. Sometimes the compiler will throw a nonsensical error, and adding explicit types helps it throw a better error.  I fixed the real issue and forgot to cleanup the explicit types. Pushed a change to clean this up.

**Pull request merged**